nuxeo-features/nuxeo-platform-comment/nuxeo-platform-comment/src/main/java/org/nuxeo/ecm/platform/comment/impl/BridgeCommentManager.java

```diff
@@ -198,4 +198,54 @@ public boolean hasFeature(Feature feature) {
             throw new UnsupportedOperationException(feature.name());
         }
     }
+
+    @Override
+    public DocumentRef getTopLevelCommentAncestor(CoreSession session, DocumentRef commentIdRef) {
+        return execute(session, commentIdRef, cm -> cm.getTopLevelCommentAncestor(session, commentIdRef));
+    }
+
+    @Override
+    public String getLocationOfCommentCreation(CoreSession session, DocumentModel documentModel) {
+        return second.getLocationOfCommentCreation(session, documentModel);
+    }
+
+    /**
+     * Executes the given function for a comment document ref, depending on the type of the provided comment manager {@link #first},
+     * {@link #second}.
+     * <p>
+     * In some cases (see the callers of this method), rely on
+     * {@link org.nuxeo.ecm.platform.comment.workflow.utils.CommentsConstants#COMMENT_PARENT_ID} is not enough, this
+     * mainly the case where {@link #first} is {@link PropertyCommentManager} and {@link #second} is
+     * {@link TreeCommentManager}, the comment storage structure is different:
+     * <ul>
+     * <li>RelationManager: Comments structures are stored in jenaGraph</li>
+     * <li>PropertyManager: All comments are stored under a hidden folder, and each comment/reply contains his direct
+     * parent in `comment:parentId`, this means if we want the commented document for example we should retrieve the
+     * `comment:parentId` of the comment thread (first comment)</li>
+     * <li>TreeCommentManager: Each comment is stored under his parent, but the first one (thread comment) is stored
+     * under a folder with `CommentRoot` type, in this case if we want to retrieve the commented document we should
+     * retrieve the `ecm:parentId` of the folder with type `CommentRoot`</li>
+     * </ul>
+     * 
+     * @since 11.1
+     */
+    protected <T> T execute(CoreSession s, DocumentRef documentRef, Function<CommentManager, T> function) {
+        DocumentModel documentModel = CoreInstance.doPrivileged(s, session -> {
+            return session.getDocument(documentRef);
+        });
+
+        if (first instanceof PropertyCommentManager && second instanceof TreeCommentManager) {
+            DocumentRef parentRef = documentModel.getParentRef();
+            if (s.getDocument(parentRef).getType().equals("HiddenFolder")) {
```

Comment:
sorry, but i didn't see the issue with this case.
if the comment is the Relation system it will root into CommentManagerImpl
but if it's already migrated into property then it will be rooted into the PropertyCommentManager otherwise the TreeCommentManager will be in charge 

thx